manifests/server/puppetserver.pp

```diff
@@ -246,10 +246,16 @@
     content => template('puppet/server/puppetserver/conf.d/puppetserver.conf.erb'),
   }
 
-  $auth_template = pick($server_auth_template, 'puppet/server/puppetserver/conf.d/auth.conf.erb')
+  if versioncmp($server_puppetserver_version, '7.0.0') >= 0 {
+    $auth_ensure = 'absent'
```

Comment:
Ensuring absent on the puppetserver auth.conf breaks puppetserver based on my testing - it really needs an auth.conf to work. The default puppetserver auth.conf that is bundled with this Puppet module seems to work with Puppetserver 7 as well.

Any particular reason to ensure absence here?

Reply:
No, this is very much a draft PR and I assumed from reading the changelog that it should be absent.

Reply:
Ok. Afaics this part can remain as it was. Only the legacy auth won't work, but that's being checked above and if it is being used catalog compilation will fail.

Reply:
My current focus is the transition to https://github.com/theforeman/puppet-puppetserver_foreman for which most has been merged now. Then I'll try to get the Puppet 7 support there. If you have patches in the mean time, I have no problem if you take over this effort or provide patches in some way. Either PRs to my repo or directly, it doesn't matter.

Reply:
This module works well enough for our [use-case](https://forge.puppet.com/modules/puppetfinland/puppetmaster) so I probably won't be needing any fixes in the near future.

One thing that could potentially help you: you may need to adjust the cipher-list parameters for PuppetDB and Puppetserver. We had to do it to support Puppetboard. See [ssl_workarounds.pp](https://github.com/Puppet-Finland/puppet-puppetmaster/blob/master/manifests/puppetboard/ssl_workarounds.pp)  for details.

Reply:
Fixed now. I deleted the wrong auth.conf. This one was the server but they meant the one that used to be shipped with the agent.